src/instructlab/model/download.py

```diff
@@ -110,22 +110,33 @@ def download_gguf(self) -> None:
             raise click.exceptions.Exit(1)
 
     def download_entire_hf_repo(self) -> None:
-        try:
-            local_dir = os.path.join(self.download_dest, self.repository)
-            os.makedirs(name=local_dir, exist_ok=True)
-
-            snapshot_download(
-                token=self.hf_token,
-                repo_id=self.repository,
-                revision=self.release,
-                local_dir=local_dir,
-            )
-        except Exception as exc:
-            click.secho(
-                f"Downloading safetensors model failed with the following Hugging Face Hub error: {exc}",
-                fg="red",
-            )
-            raise click.exceptions.Exit(1)
+        fail_count = 0
+        while True:
+            try:
+                local_dir = os.path.join(self.download_dest, self.repository)
+                os.makedirs(name=local_dir, exist_ok=True)
+
+                snapshot_download(
+                    token=self.hf_token,
+                    repo_id=self.repository,
+                    revision=self.release,
+                    local_dir=local_dir,
+                )
+            except Exception as exc:
+                # if failure was due to connection issue, retry up to 3 times
+                if ("Connection broken" in str(exc)) and (fail_count < 3):
```

Comment:
Is there a more specific exception thrown that can be used rather than using the str value?

Reply:
This didn't seem to work in the latest run